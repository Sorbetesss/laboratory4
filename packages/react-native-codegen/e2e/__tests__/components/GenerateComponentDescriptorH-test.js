/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 * @oncall react_native
 */

'use strict';

import type {GeneratorParameters} from '../../../src/generators/Utils';

const generator = require('../../../src/generators/components/GenerateComponentDescriptorH');
const {FlowParser} = require('../../../src/parsers/flow/parser');
const fs = require('fs');

const FIXTURE_DIR = `${__dirname}/../../__test_fixtures__/components`;

const fixtures = fs.readdirSync(FIXTURE_DIR);

const parser = new FlowParser();

fixtures.forEach(fixture => {
  it(`GenerateComponentDescriptorH can generate for '${fixture}'`, () => {
    const libraryName = 'RNCodegenModuleFixtures';
    const schemaType = parser.parseFile(`${FIXTURE_DIR}/${fixture}`);
    const params: GeneratorParameters = {
      libraryName: libraryName,
      schema: schemaType,
      headerPrefix: `react/renderer/components/${libraryName}/`,
    };
    const output = generator.generate(params);
    expect(Object.fromEntries(output)).toMatchSnapshot();
  });
});
