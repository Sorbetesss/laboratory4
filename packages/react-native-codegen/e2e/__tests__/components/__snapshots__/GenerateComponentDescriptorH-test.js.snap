// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenerateComponentDescriptorH can generate for 'ArrayPropsNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using ArrayPropsNativeComponentViewComponentDescriptor = ConcreteComponentDescriptor<ArrayPropsNativeComponentViewShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<ArrayPropsNativeComponentViewComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'BooleanPropNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using BooleanPropNativeComponentViewComponentDescriptor = ConcreteComponentDescriptor<BooleanPropNativeComponentViewShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<BooleanPropNativeComponentViewComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'ColorPropNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using ColorPropNativeComponentViewComponentDescriptor = ConcreteComponentDescriptor<ColorPropNativeComponentViewShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<ColorPropNativeComponentViewComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'DimensionPropNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using DimensionPropNativeComponentViewComponentDescriptor = ConcreteComponentDescriptor<DimensionPropNativeComponentViewShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<DimensionPropNativeComponentViewComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'EdgeInsetsPropNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using EdgeInsetsPropNativeComponentViewComponentDescriptor = ConcreteComponentDescriptor<EdgeInsetsPropNativeComponentViewShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<EdgeInsetsPropNativeComponentViewComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'EnumPropNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using EnumPropNativeComponentViewComponentDescriptor = ConcreteComponentDescriptor<EnumPropNativeComponentViewShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<EnumPropNativeComponentViewComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'EventNestedObjectPropsNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using EventNestedObjectPropsNativeComponentViewComponentDescriptor = ConcreteComponentDescriptor<EventNestedObjectPropsNativeComponentViewShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<EventNestedObjectPropsNativeComponentViewComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'EventPropsNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using EventPropsNativeComponentViewComponentDescriptor = ConcreteComponentDescriptor<EventPropsNativeComponentViewShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<EventPropsNativeComponentViewComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'FloatPropsNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using FloatPropsNativeComponentViewComponentDescriptor = ConcreteComponentDescriptor<FloatPropsNativeComponentViewShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<FloatPropsNativeComponentViewComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'ImagePropNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using ImagePropNativeComponentViewComponentDescriptor = ConcreteComponentDescriptor<ImagePropNativeComponentViewShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<ImagePropNativeComponentViewComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'IntegerPropNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using IntegerPropNativeComponentViewComponentDescriptor = ConcreteComponentDescriptor<IntegerPropNativeComponentViewShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<IntegerPropNativeComponentViewComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'InterfaceOnlyNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {



inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'MixedPropNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using MixedPropNativeComponentViewComponentDescriptor = ConcreteComponentDescriptor<MixedPropNativeComponentViewShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<MixedPropNativeComponentViewComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'MultiNativePropNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using MultiNativePropNativeComponentViewComponentDescriptor = ConcreteComponentDescriptor<MultiNativePropNativeComponentViewShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<MultiNativePropNativeComponentViewComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'NoPropsNoEventsNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using NoPropsNoEventsNativeComponentViewComponentDescriptor = ConcreteComponentDescriptor<NoPropsNoEventsNativeComponentViewShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<NoPropsNoEventsNativeComponentViewComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'ObjectPropsNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using ObjectPropsNativeComponentComponentDescriptor = ConcreteComponentDescriptor<ObjectPropsNativeComponentShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<ObjectPropsNativeComponentComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'PointPropNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using PointPropNativeComponentViewComponentDescriptor = ConcreteComponentDescriptor<PointPropNativeComponentViewShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<PointPropNativeComponentViewComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;

exports[`GenerateComponentDescriptorH can generate for 'StringPropNativeComponent.js' 1`] = `
Object {
  "ComponentDescriptors.h": "
/**
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
 *
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
 *
 * @generated by codegen project: GenerateComponentDescriptorH.js
 */

#pragma once

#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
#include <react/renderer/components/RNCodegenModuleFixtures/ShadowNodes.h>
#include <react/renderer/core/ConcreteComponentDescriptor.h>

namespace facebook {
namespace react {

using StringPropNativeComponentViewComponentDescriptor = ConcreteComponentDescriptor<StringPropNativeComponentViewShadowNode>;

inline void RNCodegenModuleFixtures_registerComponentDescriptors(std::shared_ptr<ComponentDescriptorProviderRegistry const> providerRegistry) {
  providerRegistry->add(concreteComponentDescriptorProvider<StringPropNativeComponentViewComponentDescriptor>());
}

} // namespace react
} // namespace facebook
",
}
`;
